### PR TITLE
Use a JsonConverter with ElementRef

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/Components/Components/src/ElementRef.cs [hunk 1]
```diff
@@ -45,13 +45,28 @@ static string CreateUniqueId()
...
+                return Guid.NewGuid().ToString("D", CultureInfo.InvariantCulture);
+            }
+        }
+
+        internal sealed class ElementRefConverter : JsonConverter<ElementRef>
+        {
+            private static readonly JsonEncodedText IdProperty = JsonEncodedText.Encode("__internalId");
+
+            public override ElementRef Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+                => throw new NotSupportedException();
```
Comment (rynowak) [inline src/Components/Components/src/ElementRef.cs]: We should add support for this while we're here, there's no good reason why this shouldn't roundtrip.

File: src/Components/Components/src/ElementRef.cs [hunk 2]
```diff
@@ -21,14 +24,11 @@ namespace Microsoft.AspNetCore.Components
         /// The Id is unique at least within the scope of a given user/circuit.
         /// This property is public to support Json serialization and should not be used by user code.
         /// </remarks>
-        [EditorBrowsable(EditorBrowsableState.Never)]
-        public string __internalId { get; }
-
-        internal string Id => __internalId;
+        internal string Id { get; }
```
Comment (rynowak) [inline src/Components/Components/src/ElementRef.cs]: nitpick: move this below the constructor?

**Comment by JamesNK** [comment]
Unit tests!

**Review by SteveSandersonMS — APPROVED** [review]
Your change is going to overlap with @rynowak's at https://github.com/aspnet/AspNetCore/pull/12264, so you might want to coordinate on that.

File: src/Components/Components/src/ElementReference.cs [hunk 3]
```diff
@@ -45,13 +45,45 @@ private static string CreateUniqueId()
...
+            public override ElementReference Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+            {
+                string id = null;
+                while (reader.Read() && reader.TokenType != JsonTokenType.EndObject)
+                {
+                    if (reader.ValueTextEquals(IdProperty.EncodedUtf8Bytes))
+                    {
+                        reader.Read();
+                        id = reader.GetString();
+                    }
```
Comment (rynowak) [inline src/Components/Components/src/ElementReference.cs]: check for unexpected properties and throw `JsonException`

**Review by JamesNK — APPROVED** [review]

**Review by rynowak — APPROVED** [review]

**Pull request merged**